src/librustc_mir/interpret/intern.rs

```diff
@@ -396,13 +420,13 @@ where
                 }
             }
         } else if ecx.memory.dead_alloc_map.contains_key(&alloc_id) {
-            // dangling pointer
-            throw_ub_format!("encountered dangling pointer in final constant")
+            // Codegen does not like dangling pointers, and generally `tcx` assumes that
+            // all allocations referenced anywhere actually exist. So, make sure we error here.
+            ecx.tcx.sess.span_err(ecx.tcx.span, "encountered dangling pointer in final constant");
```

Comment:
how about not reporting the error here, but instead reporting it in the None arm in intern_shallow, where we can easily decide not to report for references.

Reply:
We never even get to `intern_shallow` for raw pointers. So, that doesn't help to prevent codegen stumbling over dangling raw pointers.